app/controllers/users_controller.rb

```diff
@@ -0,0 +1,26 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    @user.role = 'user'
+    if @user.save
+      redirect_to @user
+    else
+      redirect_to signup_path
+    end
+  end
+
+  def show
+    @user = User.find(params[:id])
```

Comment:
in case User was not found here will be exception, you need to manage it. For example, check if user was found and redirect with message "User not found" in case not.